File: app/src/main/java/sourcecontrol/core/objects/ObjectType.java
package sourcecontrol.core.objects;

/**
 * Enum representing the different types of Git objects.
 */
public enum ObjectType {
    BLOB("blob"),
    TREE("tree"),
    COMMIT("commit"),
    TAG("tag");

    private final String typeName;

    ObjectType(String typeName) {
        this.typeName = typeName;
    }

    public String getTypeName() {
        return typeName;
    }

    public static ObjectType fromString(String type) {
        for (ObjectType objectType : values()) {
            if (objectType.typeName.equals(type)) {
                return objectType;
            }
        }
        throw new IllegalArgumentException("Unknown object type: " + type);
    }
}

File: src/core/objects/base/object-type.ts
/**
 * Enum representing the different types of Git objects.
 */
export enum ObjectType {
  BLOB = 'blob',
  TREE = 'tree',
  COMMIT = 'commit',
  TAG = 'tag',
}

export class ObjectTypeHelper {
  private constructor() {}

  public static getTypeName(type: ObjectType): string {
    return type;
  }

  public static fromString(type: string): ObjectType {
    const objectType = Object.values(ObjectType).find((t) => t === type);
    if (!objectType) {
      throw new Error(`Unknown object type: ${type}`);
    }
    return objectType;
  }
}
